src/filter/mod.rs

```diff
@@ -9,3 +9,9 @@ mod time;
 
 #[cfg(unix)]
 mod owner;
+
+pub trait Filter {
```

Comment:
I don't really get the benefit of this trait. Does this give us anything beyond having a consistent name for the filtering methods? 

Is this intended to be used more in an additional refactor?